bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Decorations.java

```diff
@@ -949,6 +921,53 @@ void setImages (Image image, Image [] images) {
 	}
 }
 
+private ImageData[] getImageData(Image[] images) {
+	ImageData[] datas = new ImageData[images.length];
+	for (int i = 0; i < images.length; i++) {
+		datas[i] = images[i].getImageData();
+	}
+	return datas;
+}
+
+private int findIndexOfClosest(ImageData[] imageData, int width, int height, int depth) {
```

Comment:
Can't this whole method be replaced by a usual `Arrays.sort` passing the `isCloserThan` implementation as a comparator? To avoid all the index calculations assuming that different arrays are sorted in the same way, you could also wrap an image with its image data into a record, so that you can execute the sort on the image data of the record while then retrieving the image stored in the first element after the sort operation.